docs/move_clusters.md

```diff
@@ -0,0 +1,59 @@
+# Moving Hive-managed Clusters
+
+## Summary
+
+There is a need to be able to move Hive-managed clusters from one Hive instance to another.
+This may be useful if the user want to replace a Hive instance with a new one or if the user
+wants to implement sharding between Hive instances.
+
+Hive will manage all of the moving for the user. The user will create a CR indicating with
+details about the destination cluster and label selectors to match against the cluster to
+move.
+
+## ClusterRelocator CRD
+
+A ClusterRelocator is cluster-scoped.
+
+This is an example of a ClusterRelocator that may be used to move all of the ClusterDeployments
+with the label `hive-instance: hive2` to the remote cluster accessed via the kubeconfig secret
+at "hive2-remote-kubecofig-secret" in the "hive" namespace.
+
+```yaml
+apiVersion: hive.openshift.io/v1
+kind: ClusterRelocator
+metadata:
+  name: hive2-cluster-relocator
+spec:
+  kubeconfigSecretRef:
+    name: hive2-remote-kubeconfig-secret
+    namespace: hive
+  clusterDeploymentSelector:
+    matchLabels:
+      hive-instance: hive2
+```
+
+## Process
+
+There is a controller that will reconcile ClusterDeployments and watch ClusterDeployments
+and ClustereRelocators. For ClusterDeployments that match with a ClusterRelocator, the
+controller will move the ClusterDeployment to the Hive instance associated with the kubeconfig
+referenced in the ClusterRelocator.
+
+1. Add the `hive.openshift.io/relocating=<relocator-name>` annotation to the ClusterDeployment. This informs other controllers not to take action on the cluster.
+This will block syncsets and machinesets from being applied and will block the clusterdeployment
+controller from running its finalizer code for the ClusterDeployment.
+1. Create a namespace (or project?) for the ClusterDeployment in the remote cluster.
+1. Copy Secrets, ConfigMaps, MachinePools, SyncSets, and SyncIdentityProviders associated
+with the ClusterDeployment to the remote cluster.
+1. Copy the ClusterDeployment to the remote cluster.
+1. Delete the ClusterDeployment.
+1. Remove the `hive.openshift.io/relocating` annotation. Replace it with a
+`hive.openshift.io/relocated=<relocator-name>` annotation. This frees the clusterdeployment
+controller to run its finalizer code for the ClusterDeployment but informs the controller
+to run perform a deprovision.
```

Comment:
"to not perform a deprovision" I assume.

Reply:
Would it be any safer to reconcile and ensure there are no finalizers set if relocated is set? (rather than acting on finalizers and checking if the annotations is set in that process) Or perhaps both.

Reply:
I do not like the idea of having two controllers that could both remove the same finalizer. I much prefer to keep the controller that adds the finalizer as the only one that removes the finalizer. Or maybe I am missing something in what you are suggesting. Would you mind elaborating a bit more?

Reply:
It could still be the same controller I think, I'm basically wondering if it's best to have the cluster deployment controller (1) detect relocated annotation being set, and thus clear the normal deprovision finalizer, or (2) to leave the normal deprovision finalizer and let that code trigger, checking if the relocated annotation is being set within. Both sounds appealing, or rather (3) if relocated is set clear finalizer, and in deprovision code which should never run, also check if relocated is set and skip or block the deprovision with lots of errors and alarm bells.

Reply:
I don't see much difference in maintainability between having a single annotation that every controller has to check and having a single function that every controller has to call.

For the purposes of example, I have changed current code like the following
```
if cd.Annotations[constants.SyncsetPauseAnnotation] == "true" {
```
to
```
if controllerutils.ShouldSyncCluster(cd, ssiLog) {
```